crates/ruff_db/src/files/path.rs

```diff
@@ -0,0 +1,176 @@
+use crate::files::{system_path_to_file, vendored_path_to_file, File};
+use crate::system::{SystemPath, SystemPathBuf};
+use crate::vendored::{VendoredPath, VendoredPathBuf};
+use crate::Db;
+
+/// Path to a file.
+///
+/// The path abstracts that files in Ruff can come from different sources:
+///
+/// * a file stored on disk
+/// * a vendored file that ships as part of the ruff binary
+/// * Future: A virtual file that references a slice of another file. For example, the CSS code in a python file.
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub enum FilePath {
```

Comment:
Maybe `FilePathBuf`, to emphasize the fact that the variants both contain owned data?

```suggestion
pub enum FilePathBuf {
```

Reply:
I prefer to only use `Owned` `Ref`, and `Buf` for types that both have an owned and unowned representation because it otherwise implies that there are two types (a ref and owned). Here, we only have an owned type.

Reply:
I see that reasoning, but here I think it incorrectly implies that this is an unsized type, since most other `*Path` types (including all others in `ruff_db`) are unsized: `std::path::Path`, `camino::Utf8Path`, `SystemPath`, `VendoredPath`, etc.

I was also planning on using this enum in `red_knot_module_resolver::path.rs` for standard-library paths. I think there's a strong possibility that doing so might mean we'd need to add a `SystemPathRef` version of this enum. (But I suppose I could do the renaming then, as part of that PR, if I do go down that road.)

Reply:
You mean a `FilePathRef`. 

We can look into this when we need it. For now we don't and I think the refactor that @carljm proposed might also remove the need for it.

Reply:
> Do mean a `FilePathRef`?

Yes! Sorry

> We can look into this when we need it

SGTM

Reply:
Though I still feel like calling it `FilePath` makes it inconsistent with `std::path::Path`, `camino::Utf8Path`, `SystemPath` and `VendoredPath`, which are all unsized types

Reply:
I agree that there's some inconsistency, but the opposite is also true: Naming it `PathBuf` implies the existence of `Path` and that would be inconsistent.

I'll resolve this. I only renamed the type from `VfsPath` to `FilePath` and I think we're over bikeshedding on the name here.